WalletWasabi/Nostr/NostrCoordinatorPublisher.cs

```diff
@@ -0,0 +1,49 @@
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin.Secp256k1;
+using NNostr.Client;
+using WalletWasabi.Bases;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Nostr;
+
+public class NostrCoordinatorPublisher : PeriodicRunner
+{
+	private readonly Uri[] _relayUris = [new("wss://relay.primal.net")];
+
+	public NostrCoordinatorPublisher(TimeSpan period, ECPrivKey key, NostrCoordinator coordinator) : base(period)
+	{
+		Coordinator = coordinator;
+		Key = key;
+		Client = NostrExtensions.Create(_relayUris, (EndPoint?)null);
+	}
+
+	private INostrClient Client { get; }
+
+	private NostrCoordinator Coordinator { get; }
+
+	private ECPrivKey Key { get; }
+
+	protected override async Task ActionAsync(CancellationToken cancel)
+	{
+		try
+		{
+			var discoveryEvent = await Key.CreateCoordinatorDiscoveryEventAsync(Coordinator).ConfigureAwait(false);
```

Comment:
Makes no sense. Is it the responsibility of a `Key` to create an Nostr event? Of course not. It is this class and no other the one that has to do it so, 
```suggestion
			var discoveryEvent = await CreateCoordinatorDiscoveryEventAsync(Coordinator).ConfigureAwait(false);
```